--- a/pw_kernel/syscall/BUILD.bazel
+++ b/pw_kernel/syscall/BUILD.bazel
@@ -14,7 +14,6 @@

 load("@bazel_skylib//lib:selects.bzl", "selects")
 load("@rules_rust//rust:defs.bzl", "rust_library")
-load("//pw_kernel:flags.bzl", "KERNEL_TEST_DEPS", "KERNEL_TEST_RUSTC_FLAGS")
 load("//pw_kernel/arch/arm_cortex_m:defs.bzl", "SUPPORTED_CORTEX_M_CPUS")

 package(default_visibility = ["//visibility:public"])
@@ -25,16 +24,11 @@
         "syscall_defs.rs",
     ],
     edition = "2024",
-    rustc_flags = KERNEL_TEST_RUSTC_FLAGS,
     tags = ["kernel"],
-    target_compatible_with = select({
-        "//pw_kernel/userspace:userspace_build_enabled": [],
-        "//conditions:default": ["@platforms//:incompatible"],
-    }),
     deps = [
         "//pw_status/rust:pw_status",
         "@rust_crates//:bitflags",
-    ] + KERNEL_TEST_DEPS,
+    ],
 )

 rust_library(
@@ -51,10 +45,6 @@
     }),
     edition = "2024",
     tags = ["kernel"],
-    target_compatible_with = select({
-        "//pw_kernel/userspace:userspace_build_enabled": [],
-        "//conditions:default": ["@platforms//:incompatible"],
-    }),
     deps = [
         ":syscall_defs",
         "//pw_status/rust:pw_status",
